### ISSUE TITLE
DistributedPubSub -  reference to get is ambiguous

### ISSUE THREAD

**Issue by arun63** [akka/akka#25849]
The project uses actor based distributed pub-sub to handle topics published and subscribed. Due to the massive workflow of the project uses akka's actor modelling. Extending of the akka cluster-tools library for topics pub-sub, I am facing the following issues from the library. 
Just for the demonstration of the issue. I have reproduced the same issue with the sample test code from the documentation.

#####  Code
I was running the following snippet of code referred from the documentation. 

```java
public class DistributedPubSubExample extends AbstractActor {

    LoggingAdapter log = Logging.getLogger(getContext().system(), this);

    /**
     * DistributedPubSubExample.
     */
    public DistributedPubSubExample() {
        ActorRef mediator =
                DistributedPubSub.get(getContext().system()).mediator();
        mediator.tell(new DistributedPubSubMediator.Subscribe("content", getSelf()),
                getSelf());
    }

    @Override
    public Receive createReceive() {
        return receiveBuilder()
                .match(String.class, msg ->
                        log.info("Got: {}", msg))
                .match(DistributedPubSubMediator.SubscribeAck.class, msg ->
                        log.info("subscribed"))
                .build();
    }
}
```
----

#####  Issue
At the time of compiling, I am getting the following error. 

```
[error] /home/arun/Developer/DistributedPubSub/src/main/java/DistributedPubSubExample.java:17:1: reference to get is ambiguous
[error]   both method get(akka.actor.ActorSystem) in akka.cluster.pubsub.DistributedPubSub and method get(akka.actor.ActorSystem) in akka.cluster.pubsub.DistributedPubSub match
[error]                 DistributedPubSub.get(getContext().system()).mediator();
[error] (Compile / compileIncremental) javac returned non-zero exit code
[error] Total time: 5 s, completed 1 Nov 2018, 11:22:03
```
-----
##### Setup
- Java 8/11 ( Tried with both the version of JVM)
- Akka 2.5.17 (same version for akka-cluster-tools) 
- scala version 2.12.7
- Build tool : sbt 1.2.6

##### Notes
- I have tried compiling the code with JRE 11 as well as JRE 1.8, the issue remains the same.

**Comment by chbatey** [comment]
Hi @arun63 - how are you switching between JDK versions? That is a known issue with Java11 that will be fixed in the next release (was fixed in Scala 2.11.7 but needs a recompile of Akka). The example in the docs is compiled as part of the build.

**Comment by arun63** [comment]
Hi @chbatey 

I am switching the JDK version from sbt build file

##### Compile using Java 11
```
javacOptions in(Compile, compile) ++= Seq("-source", "11",
  "-target", "11",
  "-g:lines")
````

##### Compile using Java 8
```
javacOptions in(Compile, compile) ++= Seq("-source", "1.8",
  "-target", "1.8",
  "-Xlint")
````

##### Info
The machine is installed with JRE 11

**Comment by johanandren** [comment]
Note that using source/target 1.8 with JDK 11 is not the same as compiling with an actual JDK 8.

**Comment by raboof** [comment]
In this case, indeed compiling your project against the jdk8 classpath will unfortunately not avoid the issue, since you are still running the resulting JDK8 classes on JDK11. I'm afraid you'll have to actually run your application with JDK8, or wait for the next Akka release (due relatively soon - no specific date yet, but likely within weeks), or apply a workaround like https://github.com/akka/akka-http/issues/2211#issuecomment-434681494

**Comment by arun63** [comment]
Hi @raboof 

Thanks for the suggestion.

The decommission from JDK 11 to JDK 8 is not a feasible solution. I look forward to next Akka release (hopefully within few weeks) to resolve it.

In the meanwhile, I have written a work-around using the scala object (as suggested).

```scala
import akka.actor.{ActorSystem, Extension, ExtensionId}

object DistributedPubSubAdaptor {

  def distributedPubSubGet[E <: Extension](extensionId: ExtensionId[E])
                                          (implicit actorSystem: ActorSystem): E =
    extensionId.get(actorSystem)

  def get(implicit actorSystem: ActorSystem): akka.cluster.pubsub.DistributedPubSub =
    distributedPubSubGet(akka.cluster.pubsub.DistributedPubSub)

}
```
-----
Using this object in the java code

```java
public DistributedPubSubExample() {
        ActorRef mediator = DistributedPubSubAdaptor.get(getContext().getSystem()).mediator();
        mediator.tell(new DistributedPubSubMediator.Subscribe("content", getSelf()),
                getSelf());
    }
```

**Comment by arun63** [comment]
Resolved in Akka version 2.5.18

**Comment by raboof** [comment]
Great, good to hear! (unfortunately it seems some IDE problems in this area remain, see #25910, but that is being worked on as well)
